File: 01-Assignment-02-02-2026/Using_C/program_18.c
/* Program 18: Print numbers from n to 0 using recursion */

#include <stdio.h>

void WriteNum(int num) {
    if (num < 0) return;
    printf("%d\n", num);
    WriteNum(num - 1);
}

int main() {
    int n ;
    printf("Give the Number : ");
    scanf("%d",&n);
    WriteNum(n);
    return 0;
}

File: 01-Assignment-02-02-2026/Using_Java/Program18.java
/*
Program 18
Java program to print the numbers from n till 0 using recursion
*/

import java.util.Scanner;

public class Program18 {
    public static int printNumber(int n){
        if (n <  0)
            return 0;
        System.out.println(n);
        return printNumber(n-1);
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        System.out.print("Enter a Positive Integer number : ");
        int Num = scanner.nextInt();
        scanner.close();
        printNumber(Num);

    }

}
